### PR TITLE
Add size of pending deletes to historical metrics

### PR DESCRIPTION
**Pull request by drcrallen**
Related to monitoring for https://github.com/druid-io/druid/issues/3283

### CODE REVIEW

File: docs/content/operations/metrics.md [hunk 1]
```diff
@@ -155,6 +155,7 @@ These metrics are for the Druid coordinator and are reset each time the coordina
 |`segment/used`|Bytes used for served segments.|dataSource, tier, priority.|< max|
 |`segment/usedPercent`|Percentage of space used by served segments.|dataSource, tier, priority.|< 100%|
 |`segment/count`|Number of served segments.|dataSource, tier, priority.|Varies.|
+|`segment/deleting`|On-disk size of segments that are waiting to be cleared out|
```
Comment (xvrl) [inline docs/content/operations/metrics.md]: in bytes I imagine?

File: server/src/main/java/io/druid/server/metrics/HistoricalMetricsMonitor.java [hunk 2]
```diff
@@ -20,35 +20,59 @@
...
+
+    for (Map.Entry<String, Long> entry : pendingDeleteSizes.entrySet()) {
+      final String dataSource = entry.getKey();
+      final long pendingDeleteSize = entry.getValue();
+      emitter.emit(
+          new ServiceMetricEvent.Builder()
+              .setDimension(DruidMetrics.DATASOURCE, dataSource)
+              .setDimension("tier", serverConfig.getTier())
+              .setDimension("priority", String.valueOf(serverConfig.getPriority()))
+              .build("segment/deleting", pendingDeleteSize)
```
Comment (xvrl) [inline server/src/main/java/io/druid/server/metrics/HistoricalMetricsMonitor.java]: I like `segment/pendingDelete` better, which also matches your variable naming :)

Comment (drcrallen) [inline server/src/main/java/io/druid/server/metrics/HistoricalMetricsMonitor.java]: fixed

File: docs/content/operations/metrics.md (hunk 1 above)
Comment (drcrallen) [inline docs/content/operations/metrics.md]: yes, added

**Comment by xvrl** [comment]
👍

File: docs/content/operations/metrics.md [hunk 3]
```diff
@@ -155,6 +155,7 @@ These metrics are for the Druid coordinator and are reset each time the coordina
 |`segment/used`|Bytes used for served segments.|dataSource, tier, priority.|< max|
 |`segment/usedPercent`|Percentage of space used by served segments.|dataSource, tier, priority.|< 100%|
 |`segment/count`|Number of served segments.|dataSource, tier, priority.|Varies.|
+|`segment/pendingDelete`|On-disk size in bytes of segments that are waiting to be cleared out|
```
Comment (drcrallen) [inline docs/content/operations/metrics.md]: whoops, missed a column here. Fixing

**Comment by gianm** [comment]
👍

**Pull request merged**